### PR TITLE
add keyword `alias_u0` with default value `false`

### PR DESCRIPTION
**Pull request by ranocha**
Cf. https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/654#discussion_r255011597 and #640.

This is my first PR changing some keyword arguments. Hence, it should be reviewed carefully because I might have missed some connections to other parts.

### CODE REVIEW

File: src/integrators/type.jl [hunk 1]
```diff
@@ -43,6 +43,7 @@ mutable struct DEOptions{absType,relType,QT,tType,F1,F2,F3,F4,F5,F6,tstopsType,d
   force_dtmin::Bool
   advance_to_tstop::Bool
   stop_at_next_tstop::Bool
+  alias_u0::Bool
```
Comment (ChrisRackauckas) [inline src/integrators/type.jl]: I'm not sure this is actually necessary since it's not used for anything.

**Comment by ChrisRackauckas** [comment]
This looks good, but we need to change the other solver packages as well to keep uniformity in the common interface.

File: src/integrators/type.jl (hunk 1 above)
Comment (ranocha) [reply src/integrators/type.jl]: I'll remove it.

**Comment by ranocha** [comment]
Is this list of packages that have to be updated complete?

- [ ] https://github.com/JuliaDiffEq/StochasticDiffEq.jl
- [ ] https://github.com/JuliaDiffEq/DelayDiffEq.jl
- [ ] https://github.com/JuliaDiffEq/DASKR.jl
- [ ] https://github.com/JuliaDiffEq/ODE.jl
- [ ] https://github.com/JuliaDiffEq/ODEInterfaceDiffEq.jl
- [ ] https://github.com/JuliaDiffEq/Sundials.jl

**Comment by ChrisRackauckas** [comment]
I think DelayDiffEq might be fine. GeometricIntegratorsDiffEq.jl and BridgeDiffEq.jl should get an update though tests are failing, and LSODA.jl should update.

**Comment by ranocha** [comment]
Okay. I've missed LSODA.jl because it's not part of JuliaDiffEq.

**Comment by ranocha** [comment]
I can't really see a possibility for aliasing for https://github.com/JuliaDiffEq/BridgeDiffEq.jl.

**Comment by ChrisRackauckas** [comment]
Yes that just uses small immutables.

**Comment by devmotion** [comment]
> I think DelayDiffEq might be fine.

Maybe it would be more intuitive to never alias the `ODEIntegrator` that is created in DelayDiffEq but depending on the keyword `alias_u0` change the `DDEIntegrator` in https://github.com/JuliaDiffEq/DelayDiffEq.jl/blob/master/src/solve.jl#L114-L117?

**Comment by ChrisRackauckas** [comment]
True that would make sense.

**Pull request merged**